src/placing-piece.js

```diff
@@ -0,0 +1,64 @@
+let gridrows = 20;
+let gridcols = 10;
+let gamegrid = Array.from({ length: gridrows }, () => Array(gridcols).fill(0));
+
+const tetrispiece = [ // creates all possible pieces and various colours
+    { shape: [[1, 1, 1, 1]], colour: 'cerulean blue' },
+    { shape: [[1, 1], [1, 1]], colour: 'honey yellow' },
+    { shape: [[1, 1, 1], [1, 0, 0]], colour: 'flurescent pink' },
+    { shape: [[1, 1, 1], [0, 0, 1]], colour: 'brick red' },
+    { shape: [[1, 1, 1], [0, 1, 0]], colour: 'mint green' },   
+    { shape: [[1, 1, 0], [0, 1, 1]], colour: 'bauge' },
+    { shape: [[0, 1, 1], [1, 1, 0]], colour: 'rose gold' }
+];
+
+function getrandompiece() { // creates a random piece and returns it
+    const randomint = Math.floor(Math.random() * tetrispiece.length);
+    return tetrispiece[randomint];
+}
+
+const Randompiece = getrandompiece(); // randomizes each piece and corresponding
+console.log("random tetris piece:", Randompiece.shape);
+console.log("random tetris colour:", Randompiece.colour);
+
+function placepiece(cols, rows, piece, gamegrid) { // places a piece in the game grid
+    for (let i = 0; i < piece.shape.length; i++) {
+        for (let j = 0; j < piece.shape[i].length; j++) {
+            if (piece.shape[i][j]) {
+                gamegrid[rows + i][cols + j] = 1; // Correctly place the piece
```

Comment:
Consider placing a character for the piece colour in the grid, this way we can use it to render the various colours!

I suggest 
`gamegrid[rows + i][cols + j] = piece.colour;`

It looks like your checks are checking "!==0" so this should work but take a quick look through to make sure it works.